CPAC/utils/serialization/core.py

```diff
@@ -0,0 +1,39 @@
+from typing import Any
+
+from nipype import __version__ as version_nipype
+from nipype.pipeline.engine import Workflow, Node
+
+from CPAC import __version__ as version_cpac
+
+WorkflowRaw = Workflow
+"""Alias for internal use. CPAC derives: CPAC.pipeline.nipype_pipeline_engine.engine.Workflow"""
+
+NodeRaw = Node
+"""Alias for internal use. CPAC derives: CPAC.pipeline.nipype_pipeline_engine.engine.Node"""
```

Comment:
Why do we want the raw classes instead of [our custom subclasses](https://github.com/nx10/C-PAC/blob/b0fb9a97776b7c4f569246723a01bd167ee4cf48/CPAC/pipeline/nipype_pipeline_engine/engine.py#L151-L698)?

Reply:
I have used the nipype version as the cpac ones are not public (not exposed via their [`__init__.py`](https://github.com/nx10/C-PAC/blob/b0fb9a97776b7c4f569246723a01bd167ee4cf48/CPAC/pipeline/nipype_pipeline_engine/__init__.py) ) 
The code is also not CPAC specific so one could argue that the base classes should be used anyway.

Reply:
The C-PAC ones are exposed in `__init__.py`:
https://github.com/FCP-INDI/C-PAC/blob/b0fb9a97776b7c4f569246723a01bd167ee4cf48/CPAC/pipeline/nipype_pipeline_engine/__init__.py#L29-L32

As to the code not being C-PAC specific, I'd argue for publishing a separate package (still under [FCP-INDI](https://github.com/FCP-INDI) probably) in that case and importing it here.

I don't know that any of the customization matters much here, and all of our `Nodes` and `Workflows` are Nipype `Nodes` and `Workflows` like all squares are rectangles, so I'm fine either way here.

(The main customizations that come to mind are wrapping strings in quotes for `.dot` file workflow representation https://github.com/FCP-INDI/C-PAC/blob/b0fb9a97776b7c4f569246723a01bd167ee4cf48/CPAC/pipeline/nipype_pipeline_engine/engine.py#L537-L538 random seeds https://github.com/FCP-INDI/C-PAC/blob/b0fb9a97776b7c4f569246723a01bd167ee4cf48/CPAC/pipeline/nipype_pipeline_engine/engine.py#L163-L164 and resource management customizations https://github.com/FCP-INDI/C-PAC/blob/b0fb9a97776b7c4f569246723a01bd167ee4cf48/CPAC/pipeline/nipype_pipeline_engine/engine.py#L158 https://github.com/FCP-INDI/C-PAC/blob/b0fb9a97776b7c4f569246723a01bd167ee4cf48/CPAC/pipeline/nipype_pipeline_engine/engine.py#L277-L279 but I guess all that would be included when the instances are initialized, right? I'm convincing myself that you're probably right to just use the base classes here.)

Reply:
Yes the class references are just used for `isinstance()` [here](https://github.com/FCP-INDI/C-PAC/blob/170893e687e7df304c7ea7259c8f8d59c99f2049/CPAC/utils/serialization/workflow_json.py#L66-L69) (which is true if it is or inherits) and for type annotations, so using them or the CPAC ones does not lead to different runtime behaviour.

I agree with splitting this off in its own python package for general nipype use. Should I do this now and cancel the PR or at a later date?

Reply:
I think this can be merged in and we can import from a separate package later whenever we split it off